pypose/module/cameras.py

```diff
@@ -0,0 +1,150 @@
+from typing import Optional
+
+import pypose
+import torch.nn
+
+
+class CamerasBase(torch.nn.Module):
+    """
+    `CamerasBase` implements a base class for all cameras.
+    For cameras, there are two different coordinate systems (or spaces)
+    - World coordinate system: This is the system the object lives - the world.
+    - Camera view coordinate system: This is the system that has its origin on
+        the camera and the Z-axis perpendicular to the image plane.
+        We assume that +X points left, and +Y points up and
+        +Z points out from the image plane.
+        The transformation from world --> view happens after applying a rotation (R)
+        and translation (T)
+    CameraBase defines methods that are common to all camera models:
+        - `camera_center` that returns the optical center of the camera in
+            world coordinates
+        - `world2camera` which returns a 3D transform from
+            world coordinates to the camera view coordinates (R, T)
+        - `world2pixel` which composes the projection
+            transform (P) with the world-to-view transform (R, T)
+
+    Args:
+        pose (LieTensor): A tensor of shape (B, *) where B is the batch size.
+    """
+
+    def __init__(self, pose=None, ):
+        super().__init__()
+        if pose is not None:
+            self.pose = pose
+        elif self.pose is None:
+            self.pose = pypose.identity_SE3()
+
+    def get_camera_center(self):
+        """
+        Returns the camera center in world coordinates.
+        Returns:
+            A tensor of shape (B, 3) where B is the batch size.
+        """
+        return self.pose.Inv().translation()
+
+    def world2camera(self, points):
+        """
+        a.k.a world to view transform.
+        Take world coordinate points and transform them to camera view coordinate.
+        Args:
+            points (torch.Tensor): A tensor of shape (B, N, 3) where B is the batch size.
+        Returns:
+            points (torch.Tensor): A tensor of shape (B, N, 3) where B is the batch size.
+        """
+        return pypose.Act(self.pose.unsqueeze(-2), points)
```

Comment:
Use `self.pose.unsqueeze(-2) @ points` could be more intuitive, but do you really need `.unsqueeze(-2)`? How about `self.pose @ points`?

Reply:
self.pose @ points doesn't work. Still need an additional dimension for broadcasting